src/components/ItemSummary.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { ItemData } from "../LibraryUtilities";
+import { LibraryContainer } from "./LibraryContainer";
+
+interface ItemSummaryProps {
+    libraryContainer: LibraryContainer,
+    data: ItemData;
+    showDescription: boolean;
+}
+
+interface ItemSummaryStates {
+    hasItemSummaryData: boolean;
+}
+
+export class ItemSummary extends React.Component<ItemSummaryProps, ItemSummaryStates> {
+    itemSummaryData: any;
+
+    constructor(props: ItemSummaryProps) {
+        super(props);
+        this.state = ({ hasItemSummaryData: false });
+        this.itemSummaryData = null;
+        this.onReceiveDataFromDynamo = this.onReceiveDataFromDynamo.bind(this);
+    }
+
+    render() {
+        this.onLibraryItemSummaryExpand();
+
+        let descriptionText = this.props.data.description;
+        let input: JSX.Element[] = [];
+        let output: JSX.Element = null;
+        let description: JSX.Element = null;
+        let icon: JSX.Element = null;
+
+        if (this.state.hasItemSummaryData) {
+            let inputParameters = this.itemSummaryData.Item1;
+            let outputParameters = this.itemSummaryData.Item2;
+            let descriptionTextFromDynamo = this.itemSummaryData.Item3;
```

Comment:
Having names like `Item1` etc. is a side-effect of serialization process of Dynamo (one of several clients), we should not allow that side-effect to leak all the way to `librarie.js`, this needs to be made something more meaningful. They need to be updated to `inputParameters`, `outputParameters` and `descriptionText`, and clients like Dynamo will have to conform to this (to be documented) format.